### PR TITLE
fix: Don't add empty lines to history #19

### PR DESCRIPTION
**Pull request by rhnvrm**
Potential fix for #19 and https://github.com/rqlite/rqlite/pull/527

### CODE REVIEW

File: term.go [hunk 1]
```diff
@@ -354,6 +354,11 @@ func (term *Terminal) prompt(buf *Buffer, in io.Reader) (string, error) {
 			continue
 		case evReturn:
 			err = buf.EndLine()
+			if buf.String() == "" {
+				term.History = term.History[:term.histIdx]
+				term.histIdx = term.histIdx - 1
+				return buf.String(), err
+			}
 			term.History[curHistIdx] = buf.String()
```
Comment (otoolep) [inline term.go]: Does it not work to simply wrap this line in a `if buf.String != ""` check?

Comment (rhnvrm) [reply term.go]: Roll back is required to handle decrementing idx as it increments it before the switch https://github.com/Bowery/prompt/blob/master/term.go#L335

**Comment by larzconwell** [comment]
Hey @rhnvrm, I went ahead and pushed a fix to master, here's the commit https://github.com/Bowery/prompt/commit/8a1d5376df1cbec3468f2138fecc44dd8b48e342, so I'm going to close this one.

Yours was mostly correct but you were modifying the wrong history item, when prompt modifies the history it only ever modifies it's latest copy(the most recent index item or `curHistIdx`) which is similar to what happens in `readline` and friends. So the change I made removed the last index from the history on enter if it's empty.

I'm gonna close this, but thanks to you both for bringing it up!

**Comment by rhnvrm** [comment]
@larzconwell @otoolep thanks for reviewing!

**Pull request closed without merging**